connection_go18.go

```diff
@@ -0,0 +1,28 @@
+// +build go1.8
+
+package mysql
+
+import (
+	"context"
+	"time"
+)
+
+func (mc *mysqlConn) Ping(ctx context.Context) error {
+	err := mc.writeCommandPacket(comPing)
+	if err != nil {
+		return err
+	}
+
+	ch := make(chan error)
+	go func() {
+		_, err := mc.readResultOK()
+		ch <- err
+	}()
```

Comment:
This will cause goroutine leak.
https://gist.github.com/methane/8da51c119193f49e990f501a44155557